src/EditorFeatures/CSharpTest/MoveDeclarationNearReference/MoveDeclarationNearReferenceTests.cs

```diff
@@ -630,5 +630,593 @@ void M()
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        // leading trivia
+        int [||]i = 5;
+        Console.WriteLine();
+
+        {
+            i = 0;
+            Console.Write(i);
+        }
+    }
+}",
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine();
+
+        {
+            // leading trivia
+            int i = 0;
+            Console.Write(i);
+        }
+    }
+}",
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMergeComments03()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        // leading trivia
+        int [||]i = 5;
+        Console.WriteLine();
+
+        // Existing trivia
+        i = 0;
+        Console.Write(i);
+    }
+}",
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine();
+
+        // leading trivia
+        // Existing trivia
+        int i = 0;
+        Console.Write(i);
+    }
+}",
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMergeComments04()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        // leading trivia
+        int [||]i = 5;
+        Console.WriteLine();
+
+        {
+            // Existing trivia
+            i = 0;
+            Console.Write(i);
+        }
+    }
+}",
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine();
+
+        {
+            // leading trivia
+            // Existing trivia
+            int i = 0;
+            Console.Write(i);
+        }
+    }
+}",
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMergeComments05()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        if (true)
+        {
+        }
+
+        // leading trivia
+        int [||]i = 5;
+        Console.WriteLine();
+
+        i = 0;
+        Console.Write(i);
+    }
+}",
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        if (true)
+        {
+        }
+
+        Console.WriteLine();
+
+        // leading trivia
+        int i = 0;
+        Console.Write(i);
+    }
+}",
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMergeComments06()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        if (true)
+        {
+        }
+
+        // leading trivia
```

Comment:
Maybe the name doesn't mean what I think it does? Lots of instances of only 1 comment in a "TestMergeComment" method.